### PR TITLE
feat: provide "index" pages for deno.land packages

### LINKED ISSUE #129: For `deno.land/x`, provide an "index" documentation page
When navigating to a "folder", we should provide an "index view" of the modules in that folder, which would list each module and documentation about its exported symbols.  So for example `https://doc.deno.land/https://deno.land/std/bytes/` would provide a listing of all the modules in `/std/bytes/` along with a brief summary of the module exports.

We can do this for `deno.land/x` and `std` because we can obtain the manifest of the files in the folder in order to determine what needs to be generated.

### PR DESCRIPTION
**Pull request by kitsonk**
Closes #129

### CODE REVIEW

**Comment by kitsonk** [comment]
I still need to write tests, but it is functionally complete and I am happy with the look.

**Comment by kitsonk** [comment]
Here is an example of what the std lib looks like now: https://heavy-goat-47.deno.dev/https://deno.land/std@0.136.0

std_–__0_136_0_–_deno_land_std___Deno_Doc

File: components/indexPage.tsx [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      linkClass = tw`hover:underline`;
+  }
+  return (
+    <tr>
+      <td class={tw`px-2 align-top`}>
+        <a href={href} class={linkClass}>{name}</a>
+      </td>
+      <td class={tw`px-2 align-top`}>
+        <MarkdownSummary>{summary}</MarkdownSummary>
+      </td>
```
Comment (bartlomieju) [inline components/indexPage.tsx:91]: Small suggestion: increase vertical padding/margin between the rows. Right now they are very dense which makes it a bit hard to scan which summary corresponds to which symbol

File: components/indexPage_test.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        path=""
+        requestUrl={requestUrl}
+        url="https://deno.land/x/tst/"
+      >
+        {indexStructure}
+      </IndexPage>,
+    )
+      .replaceAll("\n", "");
+    const expected = renderSSR(<Expected />).replaceAll("\n", "");
+    assertEquals(actual, expected);
```
Comment (bartlomieju) [inline components/indexPage_test.tsx:202]: FYI we have new [`assertSnapshot`](https://deno.land/manual@main/testing/snapshot_testing) API available which could potentially make these kind of tests nicer

**Review by bartlomieju — APPROVED** [review]
LGTM, this is a huge improvement to the current page

**Comment by ije** [comment]
nice work! small detail improvement i want to request, can we hide the border between index items? i think it looks vary rough, or you can show it only the content is expanded
Screen Shot 2022-04-26 at 20 40 56

**Comment by ry** [comment]
This looks good - but ultimately we need to move docs to deno.land itself - no long in doc.deno.land. I'm sure this work will be useful for that eventually.

**Comment by kitsonk** [comment]
> This looks good - but ultimately we need to move docs to deno.land itself - no longer on doc.deno.land. I'm sure this work will be useful for that eventually.

Yes, it is currently easier to improve doc.deno.land as we figure out how to combine the two.

**Comment by kitsonk** [comment]
> nice work! small detail improvement i want to request, can we hide the border between index items? i think it looks vary rough, or you can show it only the content is expanded 

@ije thanks for the feedback...  It was inspired by GoLang's doc website:

Standard_library_-_pkg_go_dev

We are missing a lot of module level documentation (JSDoc `@module`) in the std lib at the moment which is why there is nothing populated in the right, but when we do populate it, it will look a lot like what you see in the GoLang example, in which some sort of delimiter feels necessary to me.  Given that information, do you still feel the same way that we should eliminate it?

**Comment by kitsonk** [comment]
@ije again, thanks for the feedback, but removing it feels like the "wrong" direction personally and makes the page a lot more difficult to read:

std_–__0_136_0_–_deno_land_std___Deno_Doc_🔊

and:

oak_–__v10_4_0_–_deno_land_x___Deno_Doc_🔊

I am totally open to revisiting the whole design, as well as taking a PR that changes the design in the future, but I think we should just move forward with something with the visual line breaks until we have a better overall design.

**Pull request merged**